src/components/AddressForm.tsx

```diff
@@ -1,121 +1,150 @@
+    zip = '',
+}: AddressFormProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
-    const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [country, 'samples'], '');
+
+    let zipSampleFormat = '';
+
+    if (country && country in CONST.COUNTRY_ZIP_REGEX_DATA) {
+        const countryData = CONST.COUNTRY_ZIP_REGEX_DATA[country as keyof typeof CONST.COUNTRY_ZIP_REGEX_DATA];
+        if (countryData && 'samples' in countryData) {
+            zipSampleFormat = countryData.samples;
+        }
+    }
+
     const zipFormat = translate('common.zipCodeExampleFormat', {zipSampleFormat});
     const isUSAForm = country === CONST.COUNTRY.US;
 
     /**
-     * @param {Function} translate - translate function
-     * @param {Boolean} isUSAForm - selected country ISO code is US
-     * @param {Object} values - form input values
-     * @returns {Object} - An object containing the errors for each inputID
+     * @param translate - translate function
+     * @param isUSAForm - selected country ISO code is US
+     * @param  values - form input values
+     * @returns - An object containing the errors for each inputID
      */
-    const validator = useCallback((values) => {
-        const errors = {};
-        const requiredFields = ['addressLine1', 'city', 'country', 'state'];
+
+    const validator = useCallback((values: AddressFormValues) => {
+        type ValidatorErrors = {
+            addressLine1?: string;
+            city?: string;
+            country?: string;
+            state?: string;
+            zipPostCode?: string | Array<string | Record<string, string>>;
+        };
+
+        const errors: ValidatorErrors = {};
+        const requiredFields = ['addressLine1', 'city', 'country', 'state'] as const;
 
         // Check "State" dropdown is a valid state if selected Country is USA
-        if (values.country === CONST.COUNTRY.US && !COMMON_CONST.STATES[values.state]) {
+        if (values.country === CONST.COUNTRY.US && values.state && !(values.state in COMMON_CONST.STATES)) {
             errors.state = 'common.error.fieldRequired';
         }
 
         // Add "Field required" errors if any required field is empty
-        _.each(requiredFields, (fieldKey) => {
-            if (ValidationUtils.isRequiredFulfilled(values[fieldKey])) {
+        requiredFields.forEach((fieldKey) => {
+            const fieldValue = values[fieldKey] ?? '';
+            if (ValidationUtils.isRequiredFulfilled(fieldValue)) {
                 return;
             }
+
             errors[fieldKey] = 'common.error.fieldRequired';
         });
 
         // If no country is selected, default value is an empty string and there's no related regex data so we default to an empty object
-        const countryRegexDetails = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, values.country, {});
+        const countryRegexDetails = CONST.COUNTRY_ZIP_REGEX_DATA[values.country as keyof typeof CONST.COUNTRY_ZIP_REGEX_DATA] ?? {};
```

Comment:
Same, I guess `country` type can be updated

Reply:
We can't do it here in such a way, since we  can have an empty string as a value here (default value or changed)